src/ConfigDefinition.php

```diff
@@ -18,11 +18,11 @@ protected function getParametersDefinition(): ArrayNodeDefinition
         $parametersNode
             ->validate()
                 ->always(function ($v) {
-                    if (!empty($v['abs']) && !empty($v['s3'])) {
-                        throw new InvalidConfigurationException('Both ABS and S3 cannot be set together.');
+                    if (!empty($v['abs']) && !empty($v['s3']) && !empty($v['gcs'])) {
+                        throw new InvalidConfigurationException('GCS, ABS and S3 cannot be set together.');
                     }
-                    if (empty($v['abs']) && empty($v['s3'])) {
-                        throw new InvalidConfigurationException('ABS or S3 must be configured.');
+                    if (empty($v['abs']) && empty($v['s3']) && empty($v['gcs'])) {
+                        throw new InvalidConfigurationException('GCS, ABS or S3 must be configured.');
                     }
```

Comment:
Ta první podmínka je teď špatně, prošly by krz ní dva souběžně nastavený storage. Šlo by obě podmínky spojit do jedné, třeba:

```php
if (count(array_filter(array_map(fn($s) => !empty($v[$s]), ['abs', 's3', 'gcs']))) !== 1) {
    throw new InvalidConfigurationException('Only one of ABS, S3, or GCS needs to be configured.');
}
```

Reply:
jo ty vogo, máš pravdu - opravil jsem a doplnil i test na to

Reply:
ok je taky pryč